python-test-samples/apigw-sqs-lambda-sqs/tests/integration/test_api_gatway.py

```diff
@@ -0,0 +1,305 @@
+        response = client.get_queue_attributes(
+            QueueUrl=self.sqs_input,
+            AttributeNames=['ApproximateNumberOfMessages']
+        )
+
+    @classmethod
+    def is_validate(self,id,message,queue) -> bool:
+        """ 
+        This function will validate the id, quque name, message against the dynamodb table
+        and will do so after several seconds sleep ( due to the nature of async messaging )
+        """
+        logging.info("Validating Message id: %s", id)
+
+        f_message_found = False
+
+        for i in range(self.interval_num):
+            # Check that DynamoDB received the relevant message
+            id_items = self.dynamodb_table.query(
+                KeyConditionExpression=Key('id').eq(id))
+
+            if id_items["Count"] == 1:
+                f_message_found = True
+                break
+
+            # Sleep for interval_timeout before checking again
+            logging.info("Sleeping for %s seconds before checking again", self.interval_timeout)
+            time.sleep(self.interval_timeout)
+
+        if f_message_found == False:
+            logging.error("Message not found in DynamoDB, Hint: try to increase test timeout")
+            return False
+
+        if id_items["Items"][0]["Message"] != message:
+            logging.error("Message not matched")
+            return False
+
+        if id_items["Items"][0]["Queue_Name"] != queue:
+            logging.error("Queue Nane not matched")
+            return False
+
+        return True
+
+    def test_positive_scenario(self):
+        """
+        This test verify that a valid message is going all the way to the output queue
+        Call the API Gateway endpoint and check the response for a 200
+        If the result is 200, check in the DynamoDB that the message arrived properly
+        """
+        # Send Message to the Inbox API with Test Data, SQS SLA is 5 seconds
+        # The "type":TEST field is used, for filtering purposes,
+        # and to distinguish from other production messages
+
+        message_id = "TEST001" + self.id_postfix
+        message_text = self.test_time + " This is a test_positive_scenario"
+        message = {
+            "id": message_id,
+            "message": message_text,
+            "type": "TEST"      
+        }
+
+        response = requests.post(
+            self.api_endpoint_inbox, json=message, timeout=5)
+        self.assertEqual(response.status_code, 200)
+        logging.info("Sent message to Inbox API: %s", message)
+
+        message_queue =  self.sqs_output.split("/")[-1]
+        self.assertTrue(self.is_validate(message_id,message_text,message_queue))
+
+    def test_false_positive_scenario(self):
+        """
+        This test verify that a valid message is going all the way to the output queue
+        Call the API Gateway endpoint and check the response for a 200
+        If the result is 200, check in the DynamoDB that the message arrived properly
+        """
```

Comment:
This docstring is a copy-paste of the prior docstring and does not describe the function.

Reply:
added a distinguish sentence